// contentful-management/dist/types/adapters/REST/endpoints/ai-action.js
import copy from 'fast-copy';
import { post, del as del$1, get as get$1, put } from './raw.js';

const get = (http, params, headers) => {
    return get$1(http, `/spaces/${params.spaceId}/ai/actions/${params.aiActionId}`, {
        headers,
    });
};
const getMany = (http, params, headers) => {
    return get$1(http, `/spaces/${params.spaceId}/ai/actions`, {
        params: params.query,
        headers,
    });
};
const create = (http, params, data, headers) => {
    return post(http, `/spaces/${params.spaceId}/ai/actions`, data, { headers });
};
const update = (http, params, rawData, headers) => {
    const data = copy(rawData);
    const { sys, ...payload } = data;
    return put(http, `/spaces/${params.spaceId}/ai/actions/${params.aiActionId}`, payload, {
        headers: {
            'X-Contentful-Version': rawData.sys.version ?? 0,
            ...headers,
        },
    });
};
const del = (http, params, headers) => {
    return del$1(http, `/spaces/${params.spaceId}/ai/actions/${params.aiActionId}`, { headers });
};
const publish = (http, params, rawData, headers) => {
    return put(http, `/spaces/${params.spaceId}/ai/actions/${params.aiActionId}/published`, null, {
        headers: {
            'X-Contentful-Version': params.version,
            ...headers,
        },
    });
};
const unpublish = (http, params, headers) => {
    return del$1(http, `/spaces/${params.spaceId}/ai/actions/${params.aiActionId}/published`, {
        headers,
    });
};
const invoke = (http, params, data, headers) => {
    return post(http, `/spaces/${params.spaceId}/environments/${params.environmentId}/ai/actions/${params.aiActionId}/invoke`, data, { headers, params: params.query });
};

export { create, del, get, getMany, invoke, publish, unpublish, update };
