// contentful-management/dist/types/adapters/REST/endpoints/app-action.js
import { post, del as del$1, get as get$1, put } from './raw.js';
import { normalizeSelect } from './utils.js';

const getBaseUrl = (params) => `/organizations/${params.organizationId}/app_definitions/${params.appDefinitionId}/actions`;
const getAppActionUrl = (params) => `${getBaseUrl(params)}/${params.appActionId}`;
const getAppActionsEnvUrl = (params) => {
    if (params.environmentId) {
        return `/spaces/${params.spaceId}/environments/${params.environmentId}/actions`;
    }
    return `/spaces/${params.spaceId}/actions`;
};
const get = (http, params) => {
    return get$1(http, getAppActionUrl(params));
};
const getMany = (http, params) => {
    return get$1(http, getBaseUrl(params), {
        params: normalizeSelect(params.query),
    });
};
const getManyForEnvironment = (http, params) => {
    return get$1(http, getAppActionsEnvUrl(params), {
        params: normalizeSelect(params.query),
    });
};
const del = (http, params) => {
    return del$1(http, getAppActionUrl(params));
};
const create = (http, params, data) => {
    return post(http, getBaseUrl(params), data);
};
const update = (http, params, data) => {
    return put(http, getAppActionUrl(params), data);
};

export { create, del, get, getMany, getManyForEnvironment, update };
