// contentful-management/dist/types/adapters/REST/endpoints/bulk-action.js
import { get as get$1, post } from './raw.js';

const get = (http, params) => {
    return get$1(http, `/spaces/${params.spaceId}/environments/${params.environmentId}/bulk_actions/actions/${params.bulkActionId}`);
};
const publish = (http, params, payload) => {
    return post(http, `/spaces/${params.spaceId}/environments/${params.environmentId}/bulk_actions/publish`, payload);
};
const unpublish = (http, params, payload) => {
    return post(http, `/spaces/${params.spaceId}/environments/${params.environmentId}/bulk_actions/unpublish`, payload);
};
const validate = (http, params, payload) => {
    return post(http, `/spaces/${params.spaceId}/environments/${params.environmentId}/bulk_actions/validate`, payload);
};
const getV2 = (http, params) => {
    return get$1(http, `/spaces/${params.spaceId}/environments/${params.environmentId}/bulk_actions/${params.bulkActionId}`);
};
const publishV2 = (http, params, payload) => {
    return post(http, `/spaces/${params.spaceId}/environments/${params.environmentId}/bulk_actions`, payload);
};
const unpublishV2 = (http, params, payload) => {
    return post(http, `/spaces/${params.spaceId}/environments/${params.environmentId}/bulk_actions`, payload);
};
const validateV2 = (http, params, payload) => {
    return post(http, `/spaces/${params.spaceId}/environments/${params.environmentId}/bulk_actions`, payload);
};

export { get, getV2, publish, publishV2, unpublish, unpublishV2, validate, validateV2 };
