// contentful-management/dist/types/adapters/REST/endpoints/concept-scheme.js
import { post, put, del as del$1, get as get$1, patch as patch$1 } from './raw.js';

function basePath(orgId) {
    return `/organizations/${orgId}/taxonomy/concept-schemes`;
}
const get = (http, params) => get$1(http, `${basePath(params.organizationId)}/${params.conceptSchemeId}`);
const del = (http, params, headers) => del$1(http, `${basePath(params.organizationId)}/${params.conceptSchemeId}`, {
    headers: {
        'X-Contentful-Version': params.version,
        ...headers,
    },
});
const getMany = (http, params) => {
    const url = params.query?.pageUrl ?? basePath(params.organizationId);
    return get$1(http, url, {
        params: params.query?.pageUrl ? undefined : params.query,
    });
};
const getTotal = (http, params) => get$1(http, `${basePath(params.organizationId)}/total`);
const create = (http, params, data) => {
    return post(http, basePath(params.organizationId), data);
};
const createWithId = (http, params, data) => {
    return put(http, `${basePath(params.organizationId)}/${params.conceptSchemeId}`, data);
};
const patch = (http, params, data, headers) => {
    return patch$1(http, `${basePath(params.organizationId)}/${params.conceptSchemeId}`, data, {
        headers: {
            'X-Contentful-Version': params.version,
            'Content-Type': 'application/json-patch+json',
            ...headers,
        },
    });
};
const update = (http, params, data, headers) => {
    return put(http, `${basePath(params.organizationId)}/${params.conceptSchemeId}`, data, {
        headers: {
            'X-Contentful-Version': params.version,
            ...headers,
        },
    });
};

export { create, createWithId, del, get, getMany, getTotal, patch, update };
