// contentful-management/dist/types/adapters/REST/endpoints/environment-template.js
import copy from 'fast-copy';
import { post, del as del$1, get as get$1, put, patch } from './raw.js';

const apiPath = (organizationId, ...pathSegments) => `/organizations/${organizationId}/environment_templates/` + pathSegments.join('/');
const get = (http, { organizationId, environmentTemplateId, version, query = {} }, headers) => version
    ? get$1(http, apiPath(organizationId, environmentTemplateId, 'versions', version), {
        params: query,
        headers,
    })
    : get$1(http, apiPath(organizationId, environmentTemplateId), {
        params: query,
        headers,
    });
const getMany = (http, { organizationId, query = {} }, headers) => get$1(http, apiPath(organizationId), { params: query, headers });
const create = (http, { organizationId }, payload, headers) => post(http, apiPath(organizationId), payload, { headers });
const update = (http, { organizationId, environmentTemplateId }, payload, headers) => {
    const data = copy(payload);
    delete data.sys;
    return put(http, apiPath(organizationId, environmentTemplateId), data, {
        headers: {
            'X-Contentful-Version': payload.sys.version ?? 0,
            ...headers,
        },
    });
};
const versionUpdate = (http, { organizationId, version, environmentTemplateId }, payload, headers) => patch(http, apiPath(organizationId, environmentTemplateId, 'versions', version), payload, {
    headers,
});
const del = (http, { organizationId, environmentTemplateId }, headers) => del$1(http, apiPath(organizationId, environmentTemplateId), { headers });
const versions = (http, { organizationId, environmentTemplateId, query = {} }, headers) => get$1(http, apiPath(organizationId, environmentTemplateId, 'versions'), {
    params: query,
    headers,
});
const validate = (http, { spaceId, environmentId, environmentTemplateId, version }, payload, headers) => put(http, version
    ? `/spaces/${spaceId}/environments/${environmentId}/template_installations/${environmentTemplateId}/versions/${version}/validated`
    : `/spaces/${spaceId}/environments/${environmentId}/template_installations/${environmentTemplateId}/validated`, payload, { headers });
const install = (http, { spaceId, environmentId, environmentTemplateId }, payload, headers) => post(http, `/spaces/${spaceId}/environments/${environmentId}/template_installations/${environmentTemplateId}/versions`, payload, { headers });
const disconnect = (http, { spaceId, environmentId, environmentTemplateId }, headers) => del$1(http, `/spaces/${spaceId}/environments/${environmentId}/template_installations/${environmentTemplateId}`, { headers });

export { create, del, disconnect, get, getMany, install, update, validate, versionUpdate, versions };
