// contentful-management/dist/types/adapters/REST/endpoints/environment.js
import copy from 'fast-copy';
import { post, put, del as del$1, get as get$1 } from './raw.js';

const get = (http, params) => {
    return get$1(http, `/spaces/${params.spaceId}/environments/${params.environmentId}`);
};
const getMany = (http, params) => {
    return get$1(http, `/spaces/${params.spaceId}/environments`, {
        params: params.query,
    });
};
const update = (http, params, rawData, headers) => {
    const data = copy(rawData);
    delete data.sys;
    return put(http, `/spaces/${params.spaceId}/environments/${params.environmentId}`, data, {
        headers: {
            ...headers,
            'X-Contentful-Version': rawData.sys.version ?? 0,
        },
    });
};
const del = (http, params) => {
    return del$1(http, `/spaces/${params.spaceId}/environments/${params.environmentId}`);
};
const create = (http, params, rawData, headers) => {
    const data = copy(rawData);
    return post(http, `/spaces/${params.spaceId}/environments`, data, {
        headers,
    });
};
const createWithId = (http, params, rawData, headers) => {
    const data = copy(rawData);
    return put(http, `/spaces/${params.spaceId}/environments/${params.environmentId}`, data, {
        headers: {
            ...headers,
            ...(params.sourceEnvironmentId
                ? {
                    'X-Contentful-Source-Environment': params.sourceEnvironmentId,
                }
                : {}),
        },
    });
};

export { create, createWithId, del, get, getMany, update };
