// contentful-management/dist/types/adapters/REST/endpoints/index.js
import * as aiAction from './ai-action.js';
import * as aiActionInvocation from './ai-action-invocation.js';
import * as agent from './agent.js';
import * as agentRun from './agent-run.js';
import * as accessToken from './access-token.js';
import * as automationDefinition from './automation-definition.js';
import * as automationExecution from './automation-execution.js';
import * as apiKey from './api-key.js';
import * as appAccessToken from './app-access-token.js';
import * as appAction from './app-action.js';
import * as appActionCall from './app-action-call.js';
import * as appBundle from './app-bundle.js';
import * as appDefinition from './app-definition.js';
import * as appDetails from './app-details.js';
import * as appEventSubscription from './app-event-subscription.js';
import * as appInstallation from './app-installation.js';
import * as appKey from './app-key.js';
import * as appSignedRequest from './app-signed-request.js';
import * as appSigningSecret from './app-signing-secret.js';
import * as appUpload from './app-upload.js';
import * as asset from './asset.js';
import * as assetKey from './asset-key.js';
import * as availableLicense from './available-license.js';
import * as bulkAction from './bulk-action.js';
import * as comment from './comment.js';
import * as componentType from './component-type.js';
import * as concept from './concept.js';
import * as conceptScheme from './concept-scheme.js';
import * as contentType from './content-type.js';
import * as editorInterface from './editor-interface.js';
import * as eligibleLicense from './eligible-license.js';
import * as entry from './entry.js';
import * as environment from './environment.js';
import * as environmentAlias from './environment-alias.js';
import * as environmentTemplate from './environment-template.js';
import * as environmentTemplateInstallation from './environment-template-installation.js';
import * as extension from './extension.js';
import * as _function from './function.js';
import * as functionLog from './function-log.js';
import * as http from './http.js';
import * as locale from './locale.js';
import * as organization from './organization.js';
import * as organizationInvitation from './organization-invitation.js';
import * as organizationMembership from './organization-membership.js';
import * as oauthApplication from './oauth-application.js';
import * as personalAccessToken from './personal-access-token.js';
import * as previewApiKey from './preview-api-key.js';
import * as release from './release.js';
import * as releaseAsset from './release-asset.js';
import * as releaseEntry from './release-entry.js';
import * as releaseAction from './release-action.js';
import * as resource from './resource.js';
import * as resourceProvider from './resource-provider.js';
import * as resourceType from './resource-type.js';
import * as role from './role.js';
import * as scheduledAction from './scheduled-action.js';
import * as contentSemanticsIndex from './content-semantics-index.js';
import * as semanticDuplicates from './semantic-duplicates.js';
import * as semanticRecommendations from './semantic-recommendations.js';
import * as semanticReferenceSuggestions from './semantic-reference-suggestions.js';
import * as semanticSearch from './semantic-search.js';
import * as semanticSettings from './semantic-settings.js';
import * as snapshot from './snapshot.js';
import * as space from './space.js';
import * as spaceAddOn from './space-add-on.js';
import * as spaceMember from './space-member.js';
import * as spaceMembership from './space-membership.js';
import * as tag from './tag.js';
import * as task from './task.js';
import * as team from './team.js';
import * as teamMembership from './team-membership.js';
import * as teamSpaceMembership from './team-space-membership.js';
import * as uiConfig from './ui-config.js';
import * as upload from './upload.js';
import * as uploadCredentials from './upload-credentials.js';
import * as usage from './usage.js';
import * as user from './user.js';
import * as userUiConfig from './user-ui-config.js';
import * as webhook from './webhook.js';
import * as workflow from './workflow.js';
import * as workflowDefinition from './workflow-definition.js';
import * as workflowsChangelog from './workflows-changelog.js';

var endpoints = {
    AiAction: aiAction,
    AiActionInvocation: aiActionInvocation,
    Agent: agent,
    AgentRun: agentRun,
    ApiKey: apiKey,
    AutomationDefinition: automationDefinition,
    AutomationExecution: automationExecution,
    AppAction: appAction,
    AppActionCall: appActionCall,
    AppBundle: appBundle,
    AppDefinition: appDefinition,
    AppInstallation: appInstallation,
    AppUpload: appUpload,
    AppSignedRequest: appSignedRequest,
    AppSigningSecret: appSigningSecret,
    AppEventSubscription: appEventSubscription,
    AppKey: appKey,
    AppAccessToken: appAccessToken,
    AppDetails: appDetails,
    Asset: asset,
    AssetKey: assetKey,
    AvailableLicense: availableLicense,
    BulkAction: bulkAction,
    Comment: comment,
    ComponentType: componentType,
    Concept: concept,
    ConceptScheme: conceptScheme,
    ContentType: contentType,
    EditorInterface: editorInterface,
    EligibleLicense: eligibleLicense,
    Entry: entry,
    Environment: environment,
    EnvironmentAlias: environmentAlias,
    EnvironmentTemplate: environmentTemplate,
    EnvironmentTemplateInstallation: environmentTemplateInstallation,
    Extension: extension,
    Function: _function,
    FunctionLog: functionLog,
    Http: http,
    Locale: locale,
    Organization: organization,
    OrganizationInvitation: organizationInvitation,
    OrganizationMembership: organizationMembership,
    OAuthApplication: oauthApplication,
    PersonalAccessToken: personalAccessToken,
    AccessToken: accessToken,
    PreviewApiKey: previewApiKey,
    Release: release,
    ReleaseAsset: releaseAsset,
    ReleaseEntry: releaseEntry,
    ReleaseAction: releaseAction,
    Resource: resource,
    ResourceProvider: resourceProvider,
    ResourceType: resourceType,
    Role: role,
    ScheduledAction: scheduledAction,
    ContentSemanticsIndex: contentSemanticsIndex,
    SemanticDuplicates: semanticDuplicates,
    SemanticRecommendations: semanticRecommendations,
    SemanticReferenceSuggestions: semanticReferenceSuggestions,
    SemanticSearch: semanticSearch,
    SemanticSettings: semanticSettings,
    Snapshot: snapshot,
    Space: space,
    SpaceAddOn: spaceAddOn,
    SpaceMember: spaceMember,
    SpaceMembership: spaceMembership,
    Tag: tag,
    Task: task,
    Team: team,
    TeamMembership: teamMembership,
    TeamSpaceMembership: teamSpaceMembership,
    UIConfig: uiConfig,
    Upload: upload,
    UploadCredential: uploadCredentials,
    Usage: usage,
    User: user,
    UserUIConfig: userUiConfig,
    Webhook: webhook,
    WorkflowDefinition: workflowDefinition,
    Workflow: workflow,
    WorkflowsChangelog: workflowsChangelog,
};

export { endpoints as default };
