// contentful-management/dist/types/adapters/REST/endpoints/locale.js
import copy from 'fast-copy';
import { post, del as del$1, get as get$1, put } from './raw.js';
import { normalizeSelect } from './utils.js';

const get = (http, params) => {
    return get$1(http, `/spaces/${params.spaceId}/environments/${params.environmentId}/locales/${params.localeId}`);
};
const getMany = (http, params) => {
    return get$1(http, `/spaces/${params.spaceId}/environments/${params.environmentId}/locales`, {
        params: normalizeSelect(params.query),
    });
};
const create = (http, params, data, headers) => {
    return post(http, `/spaces/${params.spaceId}/environments/${params.environmentId}/locales`, data, {
        headers,
    });
};
const update = (http, params, rawData, headers) => {
    const data = copy(rawData);
    delete data.sys;
    delete data.default; // we should not send this back
    return put(http, `/spaces/${params.spaceId}/environments/${params.environmentId}/locales/${params.localeId}`, data, {
        headers: {
            ...headers,
            'X-Contentful-Version': rawData.sys.version ?? 0,
        },
    });
};
const del = (http, params) => {
    return del$1(http, `/spaces/${params.spaceId}/environments/${params.environmentId}/locales/${params.localeId}`);
};

export { create, del, get, getMany, update };
