// contentful-management/dist/types/adapters/REST/endpoints/raw.js
import { errorHandler } from 'contentful-sdk-core';

/* eslint-disable @typescript-eslint/no-explicit-any */
function getBaseUrl(http) {
    return http.defaults.baseURL?.split('/spaces')[0];
}
function get(http, url, config) {
    return http
        .get(url, {
        baseURL: getBaseUrl(http),
        ...config,
    })
        .then((response) => response.data, errorHandler);
}
function patch(http, url, payload, config) {
    return http
        .patch(url, payload, {
        baseURL: getBaseUrl(http),
        ...config,
    })
        .then((response) => response.data, errorHandler);
}
function post(http, url, payload, config) {
    return http
        .post(url, payload, {
        baseURL: getBaseUrl(http),
        ...config,
    })
        .then((response) => response.data, errorHandler);
}
function put(http, url, payload, config) {
    return http
        .put(url, payload, {
        baseURL: getBaseUrl(http),
        ...config,
    })
        .then((response) => response.data, errorHandler);
}
function del(http, url, config) {
    return http
        .delete(url, {
        baseURL: getBaseUrl(http),
        ...config,
    })
        .then((response) => response.data, errorHandler);
}
function http(http, url, config) {
    return http(url, {
        baseURL: getBaseUrl(http),
        ...config,
    }).then((response) => response.data, errorHandler);
}

export { del, get, http, patch, post, put };
