// contentful-management/dist/types/adapters/REST/endpoints/space.js
import copy from 'fast-copy';
import { post, del as del$1, get as get$1, put } from './raw.js';

const get = (http, params) => get$1(http, `/spaces/${params.spaceId}`, {
    params: params.include ? { include: params.include } : undefined,
});
const getMany = (http, params) => get$1(http, `/spaces`, {
    params: { ...params.query, ...(params.include ? { include: params.include } : {}) },
    headers: params.organizationId
        ? { 'X-Contentful-Organization': params.organizationId }
        : undefined,
});
const getManyForOrganization = (http, params) => get$1(http, `/organizations/${params.organizationId}/spaces`, {
    params: params.query,
});
const create = (http, params, payload, headers) => {
    return post(http, `/spaces`, payload, {
        headers: params.organizationId
            ? { ...headers, 'X-Contentful-Organization': params.organizationId }
            : headers,
    });
};
const update = (http, params, rawData, headers) => {
    const data = copy(rawData);
    delete data.sys;
    return put(http, `/spaces/${params.spaceId}`, data, {
        headers: {
            'X-Contentful-Version': rawData.sys.version ?? 0,
            ...headers,
        },
    });
};
const unarchive = (http, params, data, headers) => {
    return post(http, `/spaces/${params.spaceId}/unarchive`, data, {
        headers,
    });
};
const del = (http, params) => del$1(http, `/spaces/${params.spaceId}`);

export { create, del, get, getMany, getManyForOrganization, unarchive, update };
