// contentful-management/dist/types/adapters/REST/endpoints/team-membership.js
import copy from 'fast-copy';
import { post, del as del$1, get as get$1, put } from './raw.js';
import { normalizeSelect } from './utils.js';

const getBaseUrl = (params) => `/organizations/${params.organizationId}/teams/${params.teamId}/team_memberships`;
const getEntityUrl = (params) => `/organizations/${params.organizationId}/teams/${params.teamId}/team_memberships/${params.teamMembershipId}`;
const get = (http, params) => get$1(http, getEntityUrl(params));
const getManyForOrganization = (http, params) => get$1(http, `/organizations/${params.organizationId}/team_memberships`, {
    params: normalizeSelect(params.query),
});
const getManyForTeam = (http, params) => {
    return get$1(http, getBaseUrl(params), {
        params: normalizeSelect(params.query),
    });
};
const create = (http, params, rawData, headers) => {
    return post(http, getBaseUrl(params), rawData, { headers });
};
const update = (http, params, rawData, headers) => {
    const data = copy(rawData);
    delete data.sys;
    return put(http, getEntityUrl(params), data, {
        headers: {
            'X-Contentful-Version': rawData.sys.version || 0,
            ...headers,
        },
    });
};
const del = (http, params) => del$1(http, getEntityUrl(params));

export { create, del, get, getManyForOrganization, getManyForTeam, update };
