// contentful-management/dist/types/adapters/REST/endpoints/team-space-membership.js
import copy from 'fast-copy';
import { post, del as del$1, get as get$1, put } from './raw.js';

const getBaseUrl = (params) => `/spaces/${params.spaceId}/team_space_memberships`;
const getEntityUrl = (params) => `${getBaseUrl(params)}/${params.teamSpaceMembershipId}`;
const get = (http, params) => get$1(http, getEntityUrl(params));
const getMany = (http, params) => get$1(http, getBaseUrl(params), {
    params: params.query,
});
const getForOrganization = (http, params) => {
    return get$1(http, `/organizations/${params.organizationId}/team_space_memberships/${params.teamSpaceMembershipId}`);
};
const getManyForOrganization = (http, params) => {
    const query = params.query || {};
    if (params.teamId) {
        query['sys.team.sys.id'] = params.teamId;
    }
    return get$1(http, `/organizations/${params.organizationId}/team_space_memberships`, {
        params: params.query,
    });
};
const create = (http, params, rawData, headers) => {
    return post(http, getBaseUrl(params), rawData, {
        headers: {
            'x-contentful-team': params.teamId,
            ...headers,
        },
    });
};
const update = (http, params, rawData, headers) => {
    const data = copy(rawData);
    delete data.sys;
    return put(http, getEntityUrl(params), data, {
        headers: {
            'X-Contentful-Version': rawData.sys.version || 0,
            'x-contentful-team': rawData.sys.team.sys.id,
            ...headers,
        },
    });
};
const del = (http, params) => {
    return del$1(http, getEntityUrl(params));
};

export { create, del, get, getForOrganization, getMany, getManyForOrganization, update };
