// contentful-management/dist/types/adapters/REST/endpoints/workflow-definition.js
import copy from 'fast-copy';
import { post, del as del$1, get as get$1, put } from './raw.js';

const getBaseUrl = (params) => `/spaces/${params.spaceId}/environments/${params.environmentId}/workflow_definitions`;
const getWorkflowDefinitionUrl = (params) => `${getBaseUrl(params)}/${params.workflowDefinitionId}`;
const get = (http, params, headers) => get$1(http, getWorkflowDefinitionUrl(params), {
    headers,
});
const getMany = (http, params, headers) => get$1(http, getBaseUrl(params), {
    headers,
    params: params.query,
});
const create = (http, params, rawData, headers) => {
    const data = copy(rawData);
    return post(http, getBaseUrl(params), data, {
        headers,
    });
};
const update = (http, params, rawData, headers) => {
    const data = copy(rawData);
    delete data.sys;
    return put(http, getWorkflowDefinitionUrl(params), data, {
        headers: {
            'X-Contentful-Version': rawData.sys.version ?? 0,
            ...headers,
        },
    });
};
const del = (http, { version, ...params }, headers) => {
    return del$1(http, getWorkflowDefinitionUrl(params), {
        headers: { 'X-Contentful-Version': version, ...headers },
    });
};

export { create, del, get, getMany, update };
