// contentful-management/dist/esm/adapters/REST/endpoints/app-definition.mjs
import { post, del as del$1, get as get$1, put } from './raw.mjs';
import copy from 'fast-copy';
import { normalizeSelect, normalizeSpaceId } from './utils.mjs';

const getBaseUrl = (params) => `/organizations/${params.organizationId}/app_definitions`;
const getAppDefinitionUrl = (params) => getBaseUrl(params) + `/${params.appDefinitionId}`;
const getBaseUrlForOrgInstallations = (params) => `/app_definitions/${params.appDefinitionId}/app_installations`;
const get = (http, params) => {
    return get$1(http, getAppDefinitionUrl(params), {
        params: normalizeSelect(params.query),
    });
};
const getMany = (http, params) => {
    return get$1(http, getBaseUrl(params), {
        params: params.query,
    });
};
const create = (http, params, rawData) => {
    const data = copy(rawData);
    return post(http, getBaseUrl(params), data);
};
const update = (http, params, rawData, headers) => {
    const data = copy(rawData);
    delete data.sys;
    return put(http, getAppDefinitionUrl(params), data, {
        headers: {
            'X-Contentful-Version': rawData.sys.version ?? 0,
            ...headers,
        },
    });
};
const del = (http, params) => {
    return del$1(http, getAppDefinitionUrl(params));
};
const getInstallationsForOrg = (http, params) => {
    return get$1(http, getBaseUrlForOrgInstallations(params), {
        params: {
            ...normalizeSpaceId(normalizeSelect(params.query)),
            'sys.organization.sys.id[in]': params.organizationId,
        },
    });
};

export { create, del, get, getAppDefinitionUrl, getInstallationsForOrg, getMany, update };
