// contentful-management/dist/esm/adapters/REST/endpoints/automation-definition.mjs
import copy from 'fast-copy';
import { post, del as del$1, get as get$1, put } from './raw.mjs';

const getBaseUrl = (params) => `/spaces/${params.spaceId}/environments/${params.environmentId}/automation_definitions`;
const getAutomationDefinitionUrl = (params) => `${getBaseUrl(params)}/${params.automationDefinitionId}`;
const get = (http, params, headers) => get$1(http, getAutomationDefinitionUrl(params), {
    headers,
});
const getMany = (http, params, headers) => get$1(http, getBaseUrl(params), {
    headers,
    params: params.query,
});
const create = (http, params, rawData, headers) => {
    const data = copy(rawData);
    return post(http, getBaseUrl(params), data, {
        headers,
    });
};
const update = (http, params, rawData, headers) => {
    const data = copy(rawData);
    delete data.sys;
    return put(http, getAutomationDefinitionUrl(params), data, {
        headers: {
            'X-Contentful-Version': rawData.sys.version ?? 0,
            ...headers,
        },
    });
};
const del = (http, params, headers) => {
    return del$1(http, getAutomationDefinitionUrl(params), {
        headers: { 'X-Contentful-Version': params.version, ...headers },
    });
};

export { create, del, get, getMany, update };
