// contentful-management/dist/esm/adapters/REST/endpoints/extension.mjs
import copy from 'fast-copy';
import { post, put, del as del$1, get as get$1 } from './raw.mjs';
import { normalizeSelect } from './utils.mjs';

const getBaseUrl = (params) => `/spaces/${params.spaceId}/environments/${params.environmentId}/extensions`;
const getExtensionUrl = (params) => getBaseUrl(params) + `/${params.extensionId}`;
const get = (http, params) => {
    return get$1(http, getExtensionUrl(params), {
        params: normalizeSelect(params.query),
    });
};
const getMany = (http, params) => {
    return get$1(http, getBaseUrl(params), {
        params: normalizeSelect(params.query),
    });
};
const create = (http, params, rawData, headers) => {
    return post(http, getBaseUrl(params), rawData, { headers });
};
const createWithId = (http, params, rawData, headers) => {
    const data = copy(rawData);
    return put(http, getExtensionUrl(params), data, { headers });
};
const update = async (http, params, rawData, headers) => {
    const data = copy(rawData);
    delete data.sys;
    return put(http, getExtensionUrl(params), data, {
        headers: {
            'X-Contentful-Version': rawData.sys.version ?? 0,
            ...headers,
        },
    });
};
const del = (http, params) => {
    return del$1(http, getExtensionUrl(params));
};

export { create, createWithId, del, get, getExtensionUrl, getMany, update };
