// contentful-management/dist/esm/adapters/REST/endpoints/index.mjs
import * as aiAction from './ai-action.mjs';
import * as aiActionInvocation from './ai-action-invocation.mjs';
import * as agent from './agent.mjs';
import * as agentRun from './agent-run.mjs';
import * as accessToken from './access-token.mjs';
import * as automationDefinition from './automation-definition.mjs';
import * as automationExecution from './automation-execution.mjs';
import * as apiKey from './api-key.mjs';
import * as appAccessToken from './app-access-token.mjs';
import * as appAction from './app-action.mjs';
import * as appActionCall from './app-action-call.mjs';
import * as appBundle from './app-bundle.mjs';
import * as appDefinition from './app-definition.mjs';
import * as appDetails from './app-details.mjs';
import * as appEventSubscription from './app-event-subscription.mjs';
import * as appInstallation from './app-installation.mjs';
import * as appKey from './app-key.mjs';
import * as appSignedRequest from './app-signed-request.mjs';
import * as appSigningSecret from './app-signing-secret.mjs';
import * as appUpload from './app-upload.mjs';
import * as asset from './asset.mjs';
import * as assetKey from './asset-key.mjs';
import * as availableLicense from './available-license.mjs';
import * as bulkAction from './bulk-action.mjs';
import * as comment from './comment.mjs';
import * as componentType from './component-type.mjs';
import * as concept from './concept.mjs';
import * as conceptScheme from './concept-scheme.mjs';
import * as contentType from './content-type.mjs';
import * as editorInterface from './editor-interface.mjs';
import * as eligibleLicense from './eligible-license.mjs';
import * as entry from './entry.mjs';
import * as environment from './environment.mjs';
import * as environmentAlias from './environment-alias.mjs';
import * as environmentTemplate from './environment-template.mjs';
import * as environmentTemplateInstallation from './environment-template-installation.mjs';
import * as extension from './extension.mjs';
import * as _function from './function.mjs';
import * as functionLog from './function-log.mjs';
import * as http from './http.mjs';
import * as locale from './locale.mjs';
import * as organization from './organization.mjs';
import * as organizationInvitation from './organization-invitation.mjs';
import * as organizationMembership from './organization-membership.mjs';
import * as oauthApplication from './oauth-application.mjs';
import * as personalAccessToken from './personal-access-token.mjs';
import * as previewApiKey from './preview-api-key.mjs';
import * as release from './release.mjs';
import * as releaseAsset from './release-asset.mjs';
import * as releaseEntry from './release-entry.mjs';
import * as releaseAction from './release-action.mjs';
import * as resource from './resource.mjs';
import * as resourceProvider from './resource-provider.mjs';
import * as resourceType from './resource-type.mjs';
import * as role from './role.mjs';
import * as scheduledAction from './scheduled-action.mjs';
import * as contentSemanticsIndex from './content-semantics-index.mjs';
import * as semanticDuplicates from './semantic-duplicates.mjs';
import * as semanticRecommendations from './semantic-recommendations.mjs';
import * as semanticReferenceSuggestions from './semantic-reference-suggestions.mjs';
import * as semanticSearch from './semantic-search.mjs';
import * as semanticSettings from './semantic-settings.mjs';
import * as snapshot from './snapshot.mjs';
import * as space from './space.mjs';
import * as spaceAddOn from './space-add-on.mjs';
import * as spaceMember from './space-member.mjs';
import * as spaceMembership from './space-membership.mjs';
import * as tag from './tag.mjs';
import * as task from './task.mjs';
import * as team from './team.mjs';
import * as teamMembership from './team-membership.mjs';
import * as teamSpaceMembership from './team-space-membership.mjs';
import * as uiConfig from './ui-config.mjs';
import * as upload from './upload.mjs';
import * as uploadCredentials from './upload-credentials.mjs';
import * as usage from './usage.mjs';
import * as user from './user.mjs';
import * as userUiConfig from './user-ui-config.mjs';
import * as webhook from './webhook.mjs';
import * as workflow from './workflow.mjs';
import * as workflowDefinition from './workflow-definition.mjs';
import * as workflowsChangelog from './workflows-changelog.mjs';

var endpoints = {
    AiAction: aiAction,
    AiActionInvocation: aiActionInvocation,
    Agent: agent,
    AgentRun: agentRun,
    ApiKey: apiKey,
    AutomationDefinition: automationDefinition,
    AutomationExecution: automationExecution,
    AppAction: appAction,
    AppActionCall: appActionCall,
    AppBundle: appBundle,
    AppDefinition: appDefinition,
    AppInstallation: appInstallation,
    AppUpload: appUpload,
    AppSignedRequest: appSignedRequest,
    AppSigningSecret: appSigningSecret,
    AppEventSubscription: appEventSubscription,
    AppKey: appKey,
    AppAccessToken: appAccessToken,
    AppDetails: appDetails,
    Asset: asset,
    AssetKey: assetKey,
    AvailableLicense: availableLicense,
    BulkAction: bulkAction,
    Comment: comment,
    ComponentType: componentType,
    Concept: concept,
    ConceptScheme: conceptScheme,
    ContentType: contentType,
    EditorInterface: editorInterface,
    EligibleLicense: eligibleLicense,
    Entry: entry,
    Environment: environment,
    EnvironmentAlias: environmentAlias,
    EnvironmentTemplate: environmentTemplate,
    EnvironmentTemplateInstallation: environmentTemplateInstallation,
    Extension: extension,
    Function: _function,
    FunctionLog: functionLog,
    Http: http,
    Locale: locale,
    Organization: organization,
    OrganizationInvitation: organizationInvitation,
    OrganizationMembership: organizationMembership,
    OAuthApplication: oauthApplication,
    PersonalAccessToken: personalAccessToken,
    AccessToken: accessToken,
    PreviewApiKey: previewApiKey,
    Release: release,
    ReleaseAsset: releaseAsset,
    ReleaseEntry: releaseEntry,
    ReleaseAction: releaseAction,
    Resource: resource,
    ResourceProvider: resourceProvider,
    ResourceType: resourceType,
    Role: role,
    ScheduledAction: scheduledAction,
    ContentSemanticsIndex: contentSemanticsIndex,
    SemanticDuplicates: semanticDuplicates,
    SemanticRecommendations: semanticRecommendations,
    SemanticReferenceSuggestions: semanticReferenceSuggestions,
    SemanticSearch: semanticSearch,
    SemanticSettings: semanticSettings,
    Snapshot: snapshot,
    Space: space,
    SpaceAddOn: spaceAddOn,
    SpaceMember: spaceMember,
    SpaceMembership: spaceMembership,
    Tag: tag,
    Task: task,
    Team: team,
    TeamMembership: teamMembership,
    TeamSpaceMembership: teamSpaceMembership,
    UIConfig: uiConfig,
    Upload: upload,
    UploadCredential: uploadCredentials,
    Usage: usage,
    User: user,
    UserUIConfig: userUiConfig,
    Webhook: webhook,
    WorkflowDefinition: workflowDefinition,
    Workflow: workflow,
    WorkflowsChangelog: workflowsChangelog,
};

export { endpoints as default };
