// contentful-management/dist/esm/adapters/REST/endpoints/release.mjs
import { put, post, del as del$1, get as get$1 } from './raw.mjs';

const get = (http, params) => {
    return get$1(http, `/spaces/${params.spaceId}/environments/${params.environmentId}/releases/${params.releaseId}`);
};
const query = (http, params) => {
    // Set the schema version in the query if provided in params or query options
    const releaseSchemaVersion = params.query?.['sys.schemaVersion'] ?? params.releaseSchemaVersion ?? undefined;
    if (releaseSchemaVersion !== undefined) {
        params.query = { ...params.query, 'sys.schemaVersion': releaseSchemaVersion };
    }
    return get$1(http, `/spaces/${params.spaceId}/environments/${params.environmentId}/releases`, {
        params: params.query,
    });
};
const create = (http, params, payload) => {
    const releaseSchemaVersion = payload.sys?.schemaVersion ?? params.releaseSchemaVersion;
    if (releaseSchemaVersion === 'Release.v2') {
        payload.sys = { ...payload.sys, type: 'Release', schemaVersion: 'Release.v2' };
    }
    return post(http, `/spaces/${params.spaceId}/environments/${params.environmentId}/releases`, payload);
};
const update = (http, params, payload, headers) => {
    const releaseSchemaVersion = payload.sys?.schemaVersion ?? params.releaseSchemaVersion;
    if (releaseSchemaVersion === 'Release.v2') {
        payload.sys = { ...payload.sys, type: 'Release', schemaVersion: 'Release.v2' };
    }
    return put(http, `/spaces/${params.spaceId}/environments/${params.environmentId}/releases/${params.releaseId}`, payload, {
        headers: {
            'X-Contentful-Version': params.version,
            ...headers,
        },
    });
};
const del = (http, params) => {
    return del$1(http, `/spaces/${params.spaceId}/environments/${params.environmentId}/releases/${params.releaseId}`);
};
const publish = (http, params, headers) => {
    return put(http, `/spaces/${params.spaceId}/environments/${params.environmentId}/releases/${params.releaseId}/published`, null, {
        headers: {
            'X-Contentful-Version': params.version,
            ...headers,
        },
    });
};
const unpublish = (http, params, headers) => {
    return del$1(http, `/spaces/${params.spaceId}/environments/${params.environmentId}/releases/${params.releaseId}/published`, {
        headers: {
            'X-Contentful-Version': params.version,
            ...headers,
        },
    });
};
const validate = (http, params, payload) => {
    return post(http, `/spaces/${params.spaceId}/environments/${params.environmentId}/releases/${params.releaseId}/validate`, payload);
};
const archive = (http, params) => {
    return put(http, `/spaces/${params.spaceId}/environments/${params.environmentId}/releases/${params.releaseId}/archived`, null, {
        headers: {
            'X-Contentful-Version': params.version,
        },
    });
};
const unarchive = (http, params) => {
    return del$1(http, `/spaces/${params.spaceId}/environments/${params.environmentId}/releases/${params.releaseId}/archived`, {
        headers: {
            'X-Contentful-Version': params.version,
        },
    });
};

export { archive, create, del, get, publish, query, unarchive, unpublish, update, validate };
