// contentful-management/dist/esm/adapters/REST/endpoints/role.mjs
import copy from 'fast-copy';
import { post, put, del as del$1, get as get$1 } from './raw.mjs';
import { normalizeSelect } from './utils.mjs';

const get = (http, params) => {
    return get$1(http, `/spaces/${params.spaceId}/roles/${params.roleId}`);
};
const getMany = (http, params) => {
    return get$1(http, `/spaces/${params.spaceId}/roles`, {
        params: normalizeSelect(params.query),
    });
};
const getManyForOrganization = (http, params) => {
    return get$1(http, `/organizations/${params.organizationId}/roles`, {
        params: normalizeSelect(params.query),
    });
};
const create = (http, params, data, headers) => {
    return post(http, `/spaces/${params.spaceId}/roles`, data, {
        headers,
    });
};
const createWithId = (http, params, data, headers) => {
    return put(http, `/spaces/${params.spaceId}/roles/${params.roleId}`, data, {
        headers,
    });
};
const update = (http, params, rawData, headers) => {
    const data = copy(rawData);
    delete data.sys;
    return put(http, `/spaces/${params.spaceId}/roles/${params.roleId}`, data, {
        headers: {
            ...headers,
            'X-Contentful-Version': rawData.sys.version ?? 0,
        },
    });
};
const del = (http, params) => {
    return del$1(http, `/spaces/${params.spaceId}/roles/${params.roleId}`);
};

export { create, createWithId, del, get, getMany, getManyForOrganization, update };
