// contentful-management/dist/esm/adapters/REST/endpoints/scheduled-action.mjs
import { post, del as del$1, get as get$1, put } from './raw.mjs';
import { normalizeSelect } from './utils.mjs';

const get = (http, params) => {
    return get$1(http, `/spaces/${params.spaceId}/scheduled_actions/${params.scheduledActionId}`, {
        params: {
            'environment.sys.id': params.environmentId,
        },
    });
};
const getMany = (http, params) => {
    return get$1(http, `/spaces/${params.spaceId}/scheduled_actions`, {
        params: normalizeSelect(params.query),
    });
};
const create = (http, params, data) => {
    return post(http, `/spaces/${params.spaceId}/scheduled_actions`, data);
};
const del = (http, params) => {
    return del$1(http, `/spaces/${params.spaceId}/scheduled_actions/${params.scheduledActionId}`, {
        params: {
            'environment.sys.id': params.environmentId,
        },
    });
};
const update = (http, params, data) => {
    return put(http, `/spaces/${params.spaceId}/scheduled_actions/${params.scheduledActionId}`, data, {
        params: {
            'environment.sys.id': data.environment?.sys.id,
        },
        headers: {
            'X-Contentful-Version': params.version,
        },
    });
};

export { create, del, get, getMany, update };
