// contentful-management/dist/esm/adapters/REST/endpoints/space-membership.mjs
import copy from 'fast-copy';
import { post, put, del as del$1, get as get$1 } from './raw.mjs';

function spaceMembershipDeprecationWarning() {
    console.warn('The user attribute in the space membership root is deprecated. The attribute has been moved inside the sys  object (i.e. sys.user)');
}
const getBaseUrl = (params) => `/spaces/${params.spaceId}/space_memberships`;
const getEntityUrl = (params) => `${getBaseUrl(params)}/${params.spaceMembershipId}`;
const get = (http, params) => {
    spaceMembershipDeprecationWarning();
    return get$1(http, getEntityUrl(params));
};
const getMany = (http, params) => {
    spaceMembershipDeprecationWarning();
    return get$1(http, getBaseUrl(params), {
        params: params.query,
    });
};
const getForOrganization = (http, params) => {
    return get$1(http, `/organizations/${params.organizationId}/space_memberships/${params.spaceMembershipId}`);
};
const getManyForOrganization = (http, params) => {
    return get$1(http, `/organizations/${params.organizationId}/space_memberships`, {
        params: params.query,
    });
};
const create = (http, params, data, headers) => {
    spaceMembershipDeprecationWarning();
    return post(http, getBaseUrl(params), data, {
        headers,
    });
};
const createWithId = (http, params, data, headers) => {
    spaceMembershipDeprecationWarning();
    return put(http, getEntityUrl(params), data, {
        headers,
    });
};
const update = (http, params, rawData, headers) => {
    const data = copy(rawData);
    delete data.sys;
    return put(http, getEntityUrl(params), data, {
        headers: {
            ...headers,
            'X-Contentful-Version': rawData.sys.version ?? 0,
        },
    });
};
const del = (http, params) => {
    return del$1(http, getEntityUrl(params));
};

export { create, createWithId, del, get, getForOrganization, getMany, getManyForOrganization, update };
