// contentful-management/dist/types/entities/app-action.js
import { toPlainObject, freezeSys } from 'contentful-sdk-core';
import copy from 'fast-copy';
import { wrapCollection } from '../common-utils.js';
import enhanceWithMethods from '../enhance-with-methods.js';

/**
 * @internal
 */
function createAppActionApi(makeRequest) {
    const getParams = (data) => ({
        organizationId: data.sys.organization.sys.id,
        appDefinitionId: data.sys.appDefinition.sys.id,
        appActionId: data.sys.id,
    });
    return {
        delete: function del() {
            const data = this.toPlainObject();
            return makeRequest({
                entityType: 'AppAction',
                action: 'delete',
                params: getParams(data),
            });
        },
    };
}
/**
 * @internal
 * @param makeRequest - function to make requests via an adapter
 * @param data - Raw App Bundle data
 * @returns Wrapped App Bundle data
 */
function wrapAppAction(makeRequest, data) {
    const appAction = toPlainObject(copy(data));
    const appActionWithMethods = enhanceWithMethods(appAction, createAppActionApi(makeRequest));
    return freezeSys(appActionWithMethods);
}
/**
 * @internal
 * @param makeRequest - function to make requests via an adapter
 * @param data - Raw App Bundle collection data
 * @returns Wrapped App Bundle collection data
 */
const wrapAppActionCollection = wrapCollection(wrapAppAction);

export { wrapAppAction, wrapAppActionCollection };
