// contentful-management/dist/types/entities/app-key.js
import copy from 'fast-copy';
import { toPlainObject } from 'contentful-sdk-core';
import enhanceWithMethods from '../enhance-with-methods.js';
import { wrapCollection } from '../common-utils.js';

function createKeyApi(makeRequest) {
    const getParams = (data) => ({
        organizationId: data.sys.organization.sys.id,
        appDefinitionId: data.sys.appDefinition.sys.id,
        fingerprint: data.sys.id,
    });
    return {
        delete: function del() {
            const self = this;
            return makeRequest({
                entityType: 'AppKey',
                action: 'delete',
                params: getParams(self),
            });
        },
    };
}
/**
 * @internal
 * @param http - HTTP client instance
 * @param data - Raw AppKey data
 * @returns Wrapped AppKey data
 */
function wrapAppKey(makeRequest, data) {
    const key = toPlainObject(copy(data));
    return enhanceWithMethods(key, createKeyApi(makeRequest));
}
/**
 * @internal
 * @param makeRequest - function to make requests via an adapter
 * @param data - Raw App Key collection data
 * @returns Wrapped App Key collection data
 */
const wrapAppKeyCollection = wrapCollection(wrapAppKey);

export { wrapAppKey, wrapAppKeyCollection };
