// contentful-management/dist/types/entities/app-upload.js
import { toPlainObject, freezeSys } from 'contentful-sdk-core';
import copy from 'fast-copy';
import enhanceWithMethods from '../enhance-with-methods.js';

/**
 * @internal
 */
function createAppUploadApi(makeRequest) {
    const getParams = (data) => ({
        organizationId: data.sys.organization.sys.id,
        appUploadId: data.sys.id,
    });
    return {
        delete: function del() {
            const data = this.toPlainObject();
            return makeRequest({
                entityType: 'AppUpload',
                action: 'delete',
                params: getParams(data),
            });
        },
    };
}
/**
 * @internal
 * @param makeRequest - function to make requests via an adapter
 * @param data - Raw App Upload data
 * @returns Wrapped App Upload data
 */
function wrapAppUpload(makeRequest, data) {
    const appUpload = toPlainObject(copy(data));
    const appUploadWithMethods = enhanceWithMethods(appUpload, createAppUploadApi(makeRequest));
    return freezeSys(appUploadWithMethods);
}

export { wrapAppUpload };
