// contentful-management/dist/types/entities/locale.js
import copy from 'fast-copy';
import { toPlainObject, freezeSys } from 'contentful-sdk-core';
import enhanceWithMethods from '../enhance-with-methods.js';
import { wrapCollection } from '../common-utils.js';

/**
 * @internal
 */
function createLocaleApi(makeRequest) {
    const getParams = (locale) => ({
        spaceId: locale.sys.space.sys.id,
        environmentId: locale.sys.environment.sys.id,
        localeId: locale.sys.id,
    });
    return {
        update: function () {
            const raw = this.toPlainObject();
            return makeRequest({
                entityType: 'Locale',
                action: 'update',
                params: getParams(raw),
                payload: raw,
            }).then((data) => wrapLocale(makeRequest, data));
        },
        delete: function () {
            const raw = this.toPlainObject();
            return makeRequest({
                entityType: 'Locale',
                action: 'delete',
                params: getParams(raw),
            }).then(() => {
                // noop
            });
        },
    };
}
/**
 * @internal
 * @param makeRequest - function to make requests via an adapter
 * @param data - Raw locale data
 * @returns Wrapped locale data
 */
function wrapLocale(makeRequest, data) {
    delete data.internal_code;
    const locale = toPlainObject(copy(data));
    const localeWithMethods = enhanceWithMethods(locale, createLocaleApi(makeRequest));
    return freezeSys(localeWithMethods);
}
/**
 * @internal
 */
const wrapLocaleCollection = wrapCollection(wrapLocale);

export { wrapLocale, wrapLocaleCollection };
