// contentful-management/dist/types/entities/personal-access-token.js
import copy from 'fast-copy';
import { toPlainObject, freezeSys } from 'contentful-sdk-core';
import enhanceWithMethods from '../enhance-with-methods.js';
import { wrapCollection } from '../common-utils.js';

/**
 * @internal
 * @param makeRequest - function to make requests via an adapter
 * @param data - Raw  personal access token data
 * @returns Wrapped personal access token
 */
function wrapPersonalAccessToken(makeRequest, data) {
    const personalAccessToken = toPlainObject(copy(data));
    const personalAccessTokenWithMethods = enhanceWithMethods(personalAccessToken, {
        revoke: function () {
            return makeRequest({
                entityType: 'PersonalAccessToken',
                action: 'revoke',
                params: { tokenId: data.sys.id },
            }).then((data) => wrapPersonalAccessToken(makeRequest, data));
        },
    });
    return freezeSys(personalAccessTokenWithMethods);
}
/**
 * @internal
 * @param makeRequest - function to make requests via an adapter
 * @param data - Raw personal access collection data
 * @returns Wrapped personal access token collection data
 */
const wrapPersonalAccessTokenCollection = wrapCollection(wrapPersonalAccessToken);

export { wrapPersonalAccessToken, wrapPersonalAccessTokenCollection };
