// contentful-management/dist/types/entities/role.js
import copy from 'fast-copy';
import { toPlainObject, freezeSys } from 'contentful-sdk-core';
import enhanceWithMethods from '../enhance-with-methods.js';
import { wrapCollection } from '../common-utils.js';

/**
 * @internal
 */
function createRoleApi(makeRequest) {
    const getParams = (data) => ({
        spaceId: data.sys.space.sys.id,
        roleId: data.sys.id,
    });
    return {
        update: function update() {
            const data = this.toPlainObject();
            return makeRequest({
                entityType: 'Role',
                action: 'update',
                params: getParams(data),
                payload: data,
            }).then((data) => wrapRole(makeRequest, data));
        },
        delete: function del() {
            const data = this.toPlainObject();
            return makeRequest({
                entityType: 'Role',
                action: 'delete',
                params: getParams(data),
            });
        },
    };
}
/**
 * @internal
 * @param makeRequest - function to make requests via an adapter
 * @param data - Raw role data
 * @returns Wrapped role data
 */
function wrapRole(makeRequest, data) {
    const role = toPlainObject(copy(data));
    const roleWithMethods = enhanceWithMethods(role, createRoleApi(makeRequest));
    return freezeSys(roleWithMethods);
}
/**
 * @internal
 */
const wrapRoleCollection = wrapCollection(wrapRole);

export { wrapRole, wrapRoleCollection };
