// contentful-management/dist/types/entities/space-membership.js
import copy from 'fast-copy';
import { toPlainObject, freezeSys } from 'contentful-sdk-core';
import enhanceWithMethods from '../enhance-with-methods.js';
import { wrapCollection } from '../common-utils.js';

/**
 * @internal
 */
function createSpaceMembershipApi(makeRequest) {
    const getParams = (data) => ({
        spaceId: data.sys.space.sys.id,
        spaceMembershipId: data.sys.id,
    });
    return {
        update: function update() {
            const data = this.toPlainObject();
            return makeRequest({
                entityType: 'SpaceMembership',
                action: 'update',
                params: getParams(data),
                payload: data,
            }).then((data) => wrapSpaceMembership(makeRequest, data));
        },
        delete: function del() {
            const data = this.toPlainObject();
            return makeRequest({
                entityType: 'SpaceMembership',
                action: 'delete',
                params: getParams(data),
            });
        },
    };
}
/**
 * @internal
 * @param makeRequest - function to make requests via an adapter
 * @param data - Raw space membership data
 * @returns Wrapped space membership data
 */
function wrapSpaceMembership(makeRequest, data) {
    const spaceMembership = toPlainObject(copy(data));
    const spaceMembershipWithMethods = enhanceWithMethods(spaceMembership, createSpaceMembershipApi(makeRequest));
    return freezeSys(spaceMembershipWithMethods);
}
/**
 * @internal
 */
const wrapSpaceMembershipCollection = wrapCollection(wrapSpaceMembership);

export { wrapSpaceMembership, wrapSpaceMembershipCollection };
