// contentful-management/dist/types/entities/team-space-membership.js
import copy from 'fast-copy';
import { toPlainObject, freezeSys } from 'contentful-sdk-core';
import enhanceWithMethods from '../enhance-with-methods.js';
import { wrapCollection } from '../common-utils.js';

/**
 * @internal
 */
function createTeamSpaceMembershipApi(makeRequest) {
    const getParams = (data) => ({
        teamSpaceMembershipId: data.sys.id,
        spaceId: data.sys.space.sys.id,
    });
    return {
        update: function () {
            const raw = this.toPlainObject();
            return makeRequest({
                entityType: 'TeamSpaceMembership',
                action: 'update',
                params: getParams(raw),
                payload: raw,
            }).then((data) => wrapTeamSpaceMembership(makeRequest, data));
        },
        delete: function del() {
            const data = this.toPlainObject();
            return makeRequest({
                entityType: 'TeamSpaceMembership',
                action: 'delete',
                params: getParams(data),
            });
        },
    };
}
/**
 * @internal
 * @param makeRequest - function to make requests via an adapter
 * @param data - Raw space membership data
 * @returns Wrapped team space membership data
 */
function wrapTeamSpaceMembership(makeRequest, data) {
    const teamSpaceMembership = toPlainObject(copy(data));
    const teamSpaceMembershipWithMethods = enhanceWithMethods(teamSpaceMembership, createTeamSpaceMembershipApi(makeRequest));
    return freezeSys(teamSpaceMembershipWithMethods);
}
/**
 * @internal
 */
const wrapTeamSpaceMembershipCollection = wrapCollection(wrapTeamSpaceMembership);

export { wrapTeamSpaceMembership, wrapTeamSpaceMembershipCollection };
