// contentful-management/dist/esm/constants/editor-interface-defaults/editors-defaults.mjs
import { WidgetNamespace, DEFAULT_EDITOR_ID } from './types.mjs';

const EntryEditorWidgetTypes = {
    DEFAULT_EDITOR: {
        name: 'Editor',
        id: DEFAULT_EDITOR_ID},
    REFERENCE_TREE: {
        name: 'References',
        id: 'reference-tree'},
    TAGS_EDITOR: {
        name: 'Tags',
        id: 'tags-editor'},
};
const DefaultEntryEditor = {
    widgetId: EntryEditorWidgetTypes.DEFAULT_EDITOR.id,
    widgetNamespace: WidgetNamespace.EDITOR_BUILTIN,
    name: EntryEditorWidgetTypes.DEFAULT_EDITOR.name,
};
const ReferencesEntryEditor = {
    widgetId: EntryEditorWidgetTypes.REFERENCE_TREE.id,
    widgetNamespace: WidgetNamespace.EDITOR_BUILTIN,
    name: EntryEditorWidgetTypes.REFERENCE_TREE.name,
};
const TagsEditor = {
    widgetId: EntryEditorWidgetTypes.TAGS_EDITOR.id,
    widgetNamespace: WidgetNamespace.EDITOR_BUILTIN,
    name: EntryEditorWidgetTypes.TAGS_EDITOR.name,
};
const EntryConfiguration = [DefaultEntryEditor, ReferencesEntryEditor, TagsEditor];

export { EntryConfiguration, EntryEditorWidgetTypes };
