// contentful-management/dist/esm/entities/agent.mjs
import { toPlainObject, freezeSys } from 'contentful-sdk-core';
import copy from 'fast-copy';
import { wrapCollection } from '../common-utils.mjs';
import enhanceWithMethods from '../enhance-with-methods.mjs';
import { wrapAgentGenerateResponse } from './agent-run.mjs';

function createAgentApi(makeRequest) {
    const getParams = (data) => ({
        spaceId: data.sys.space.sys.id,
        environmentId: data.sys.environment.sys.id,
        agentId: data.sys.id,
    });
    return {
        generate: function generate(payload) {
            const self = this;
            return makeRequest({
                entityType: 'Agent',
                action: 'generate',
                params: getParams(self),
                payload,
            }).then((data) => wrapAgentGenerateResponse(makeRequest, data));
        },
    };
}
function wrapAgent(makeRequest, data) {
    const agent = toPlainObject(copy(data));
    const agentWithMethods = enhanceWithMethods(agent, createAgentApi(makeRequest));
    return freezeSys(agentWithMethods);
}
const wrapAgentCollection = wrapCollection(wrapAgent);

export { wrapAgent, wrapAgentCollection };
