// contentful-management/dist/esm/entities/app-details.mjs
import copy from 'fast-copy';
import { toPlainObject } from 'contentful-sdk-core';
import enhanceWithMethods from '../enhance-with-methods.mjs';

/**
 * @internal
 * @param makeRequest - function to make requests via an adapter
 * @returns Wrapped App Details data
 */
function createAppDetailsApi(makeRequest) {
    const getParams = (data) => ({
        organizationId: data.sys.organization.sys.id,
        appDefinitionId: data.sys.appDefinition.sys.id,
    });
    return {
        delete: function del() {
            const self = this;
            return makeRequest({
                entityType: 'AppDetails',
                action: 'delete',
                params: getParams(self),
            });
        },
    };
}
/**
 * @internal
 * @param http - HTTP client instance
 * @param data - Raw AppDetails data
 * @returns Wrapped AppDetails data
 */
function wrapAppDetails(makeRequest, data) {
    const appDetails = toPlainObject(copy(data));
    return enhanceWithMethods(appDetails, createAppDetailsApi(makeRequest));
}

export { wrapAppDetails };
