// contentful-management/dist/esm/entities/app-event-subscription.mjs
import copy from 'fast-copy';
import { toPlainObject } from 'contentful-sdk-core';
import enhanceWithMethods from '../enhance-with-methods.mjs';

function createEventSubscriptionApi(makeRequest) {
    const getParams = (data) => ({
        organizationId: data.sys.organization.sys.id,
        appDefinitionId: data.sys.appDefinition.sys.id,
    });
    return {
        delete: function del() {
            const self = this;
            return makeRequest({
                entityType: 'AppEventSubscription',
                action: 'delete',
                params: getParams(self),
            });
        },
    };
}
/**
 * @internal
 * @param http - HTTP client instance
 * @param data - Raw AppEventSubscription data
 * @returns Wrapped AppEventSubscription data
 */
function wrapAppEventSubscription(makeRequest, data) {
    const eventSubscription = toPlainObject(copy(data));
    return enhanceWithMethods(eventSubscription, createEventSubscriptionApi(makeRequest));
}

export { wrapAppEventSubscription };
