// contentful-management/dist/esm/entities/environment-template-installation.mjs
import { toPlainObject, freezeSys } from 'contentful-sdk-core';
import copy from 'fast-copy';
import { wrapCursorPaginatedCollection } from '../common-utils.mjs';

var EnvironmentTemplateInstallationStatuses;
(function (EnvironmentTemplateInstallationStatuses) {
    EnvironmentTemplateInstallationStatuses["created"] = "created";
    EnvironmentTemplateInstallationStatuses["inProgress"] = "inProgress";
    EnvironmentTemplateInstallationStatuses["failed"] = "failed";
    EnvironmentTemplateInstallationStatuses["succeeded"] = "succeeded";
    EnvironmentTemplateInstallationStatuses["disconnected"] = "disconnected";
    EnvironmentTemplateInstallationStatuses["inRetry"] = "inRetry";
})(EnvironmentTemplateInstallationStatuses || (EnvironmentTemplateInstallationStatuses = {}));
function wrapEnvironmentTemplateInstallation(makeRequest, data) {
    const environmentTemplate = toPlainObject(copy(data));
    return freezeSys(environmentTemplate);
}
const wrapEnvironmentTemplateInstallationCollection = wrapCursorPaginatedCollection(wrapEnvironmentTemplateInstallation);

export { wrapEnvironmentTemplateInstallation, wrapEnvironmentTemplateInstallationCollection };
