// contentful-management/dist/esm/entities/extension.mjs
import copy from 'fast-copy';
import { toPlainObject, freezeSys } from 'contentful-sdk-core';
import enhanceWithMethods from '../enhance-with-methods.mjs';
import { wrapCollection } from '../common-utils.mjs';

/**
 * @internal
 */
function createExtensionApi(makeRequest) {
    const getParams = (data) => ({
        spaceId: data.sys.space.sys.id,
        environmentId: data.sys.environment.sys.id,
        extensionId: data.sys.id,
    });
    return {
        update: function update() {
            const data = this.toPlainObject();
            return makeRequest({
                entityType: 'Extension',
                action: 'update',
                params: getParams(data),
                payload: data,
            }).then((response) => wrapExtension(makeRequest, response));
        },
        delete: function del() {
            const data = this.toPlainObject();
            return makeRequest({
                entityType: 'Extension',
                action: 'delete',
                params: getParams(data),
            });
        },
    };
}
/**
 * @internal
 * @param makeRequest - function to make requests via an adapter
 * @param data - Raw UI Extension data
 * @returns Wrapped UI Extension data
 */
function wrapExtension(makeRequest, data) {
    const extension = toPlainObject(copy(data));
    const extensionWithMethods = enhanceWithMethods(extension, createExtensionApi(makeRequest));
    return freezeSys(extensionWithMethods);
}
/**
 * @internal
 */
const wrapExtensionCollection = wrapCollection(wrapExtension);

export { wrapExtension, wrapExtensionCollection };
