// contentful-management/dist/esm/entities/workflow-definition.mjs
import 'contentful-sdk-core';
import 'fast-copy';

var WorkflowStepPermissionType;
(function (WorkflowStepPermissionType) {
    WorkflowStepPermissionType["EntityPermission"] = "entity_permission";
    WorkflowStepPermissionType["WorkflowPermission"] = "workflow_permission";
})(WorkflowStepPermissionType || (WorkflowStepPermissionType = {}));
var WorkflowStepPermissionAction;
(function (WorkflowStepPermissionAction) {
    WorkflowStepPermissionAction["Edit"] = "edit";
    WorkflowStepPermissionAction["Publish"] = "publish";
    WorkflowStepPermissionAction["Delete"] = "delete";
})(WorkflowStepPermissionAction || (WorkflowStepPermissionAction = {}));
var WorkflowStepPermissionEffect;
(function (WorkflowStepPermissionEffect) {
    WorkflowStepPermissionEffect["Allow"] = "allow";
    WorkflowStepPermissionEffect["Deny"] = "deny";
})(WorkflowStepPermissionEffect || (WorkflowStepPermissionEffect = {}));
/* Workflow Step Action */
var WorkflowStepActionType;
(function (WorkflowStepActionType) {
    WorkflowStepActionType["App"] = "app";
    WorkflowStepActionType["Email"] = "email";
    WorkflowStepActionType["Task"] = "task";
})(WorkflowStepActionType || (WorkflowStepActionType = {}));

export { WorkflowStepActionType, WorkflowStepPermissionAction, WorkflowStepPermissionEffect, WorkflowStepPermissionType };
